backend/src/main/java/com/redhat/cloud/notifications/db/repositories/DrawerNotificationRepository.java

```diff
@@ -0,0 +1,138 @@
+package com.redhat.cloud.notifications.db.repositories;
+
+import com.redhat.cloud.notifications.db.Query;
+import com.redhat.cloud.notifications.models.DrawerEntry;
+import com.redhat.cloud.notifications.models.DrawerNotification;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class DrawerNotificationRepository {
+
+    @Inject
+    EntityManager entityManager;
+
+    @Transactional
+    public Integer updateReadStatus(String orgId, String username, Set<UUID> notificationIds, Boolean readStatus) {
+
+        String hql = "UPDATE DrawerNotification SET read = :readStatus "
+            + "WHERE orgId = :orgId and userId = :userId and id in (:notificationIds)";
+
+        return entityManager.createQuery(hql)
+            .setParameter("orgId", orgId)
+            .setParameter("userId", username)
+            .setParameter("readStatus", readStatus)
+            .setParameter("notificationIds", notificationIds)
+            .executeUpdate();
+    }
+
+    public List<DrawerEntry> getNotifications(String orgId, String username, Set<UUID> bundleIds, Set<UUID> appIds, Set<UUID> eventTypeIds,
+                                              LocalDateTime startDate, LocalDateTime endDate, Boolean readStatus, Query query) {
+        query.setSortFields(DrawerNotification.SORT_FIELDS);
+        query.setDefaultSortBy("created:DESC");
+        Optional<Query.Sort> sort = query.getSort();
+
+        String hql = "SELECT dn.id, dn.orgId, dn.userId, dn.read, " +
+            "dn.event.bundleDisplayName, dn.event.applicationDisplayName, dn.event.eventTypeDisplayName, dn.event.created "
+            + "FROM DrawerNotification dn where dn.orgId = :orgId and dn.userId = :userid";
+
+        hql = addHqlConditions(hql, bundleIds, appIds, eventTypeIds, startDate, endDate, readStatus);
+        if (sort.isPresent()) {
+            hql += getOrderBy(sort.get());
+        }
+
+        TypedQuery<Object[]> typedQuery = entityManager.createQuery(hql, Object[].class);
```

Comment:
If there any reason why we can't do this?
```suggestion
        TypedQuery<DrawerNotification> typedQuery = entityManager.createQuery("SELECT dn FROM DrawerNotification dn WHERE ...", DrawerNotification.class);
```

Reply:
Please ignore my comment. I'm resolving this.